core/src/test/scala/org/apache/spark/deploy/SparkSubmitSuite.scala

```diff
@@ -74,20 +74,27 @@ trait TestPrematureExit {
     @volatile var exitedCleanly = false
     mainObject.exitFn = (_) => exitedCleanly = true
 
+    var exception: Exception = null
     val thread = new Thread {
       override def run() = try {
         mainObject.main(input)
       } catch {
         // If exceptions occur after the "exit" has happened, fine to ignore them.
         // These represent code paths not reachable during normal execution.
-        case e: Exception => if (!exitedCleanly) throw e
+        case e: Exception => exception = e
       }
     }
     thread.start()
     thread.join()
     val joined = printStream.lineBuffer.mkString("\n")
-    if (!joined.contains(searchString)) {
-      fail(s"Search string '$searchString' not found in $joined")
+    val searchStrContainsInEx = exception != null && exception.getMessage.contains(searchString)
+    if(!searchStrContainsInEx){
```

Comment:
You need spaces around parens here. But I think there's a logic issue; if exception is `null` but `exitedCleanly` is `false`, you try to throw `null`. I get that it won't be null if so, but that highlights that I think they're redundant. And I don't think it's helping to pull out the condition here in a variable.

I think this is just simpler to delete `exitedCleanly` and ...

```
@volatile var: Exception = null
...
if (exception != null) {
  if (!exception.getMessage.contains(searchString)) {
    throw exception
  }
} else if (!joined.contains(searchString) {
  fail(...)
}
```
I think that's equivalent?

Reply:
I've update as per this, thanks for the suggestion.